### PR TITLE
rubocop: fix some rubocop offences

### PR DESCRIPTION
**Pull request by javian**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [X] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [X] Have you successfully run `brew style` with your changes locally?
- [X] Have you successfully run `brew tests` with your changes locally?

-----
Fixes a couple of the rubocop offences currently in the tests. First batch of trying to wear down #4962.

### CODE REVIEW

File: Library/Homebrew/test/.rubocop_todo.yml [hunk 1]
```diff
@@ -121,17 +121,6 @@ RSpec/RepeatedDescription:
     - 'rubocops/lines_cop_spec.rb'
     - 'tab_spec.rb'
 
-# Offense count: 10
-RSpec/RepeatedExample:
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/.rubocop_todo.yml]: Are these examples with exactly the same code or just the same `it`/name?

Comment (reitermarkus) [reply Library/Homebrew/test/.rubocop_todo.yml]: These are not repeated, they are using a different setup, so I'm not sure what this is flagging.

Comment (MikeMcQuaid) [reply Library/Homebrew/test/.rubocop_todo.yml]: Not the same `it` string?

Comment (reitermarkus) [reply Library/Homebrew/test/.rubocop_todo.yml]: I don't think so. For example, one has “Sh“ and one has ”Bash“ at the end.

Comment (reitermarkus) [reply Library/Homebrew/test/.rubocop_todo.yml]: Oh, is is indeed the same exact code as the one before …

File: Library/Homebrew/test/utils/shell_spec.rb [hunk 2]
```diff
@@ -7,11 +7,6 @@
       expect(subject.profile).to eq("~/.bash_profile")
     end
 
-    it "returns ~/.bash_profile for Sh" do
-      ENV["SHELL"] = "/bin/another_shell"
```
Comment (reitermarkus) [inline Library/Homebrew/test/utils/shell_spec.rb]: Change this to `/bin/sh` instead.

File: Library/Homebrew/test/.rubocop_todo.yml (hunk 1 above)
Comment (javian) [reply Library/Homebrew/test/.rubocop_todo.yml]: Yes, unless I've misunderstood something these should all be duplicate tests that I've removed.

File: Library/Homebrew/test/utils/shell_spec.rb (hunk 2 above)
Comment (javian) [reply Library/Homebrew/test/utils/shell_spec.rb]: Will do.

File: Library/Homebrew/test/.rubocop_todo.yml (hunk 1 above)
Comment (javian) [reply Library/Homebrew/test/.rubocop_todo.yml]: The code is (or was) duplicated with different `it` strings.

Comment (MikeMcQuaid) [reply Library/Homebrew/test/.rubocop_todo.yml]: You haven't misunderstood, you're correct 👍

**Comment by MikeMcQuaid** [comment]
Thanks again @javian!

**Pull request merged**